### PR TITLE
Hub edits

### PR DESCRIPTION
**Pull request by Loquacity**
https://bugzilla.suse.com/show_bug.cgi?id=1171389
close: https://github.com/SUSE/spacewalk/issues/11403

### CODE REVIEW

**Comment by Loquacity** [comment]
> LGTM, but it does not address the problem of te brackets in the python script (see bugzilla issue).

Yep, I think I need a SME. Who would be good, do you think?

**Comment by Bischoff** [comment]
> > LGTM, but it does not address the problem of te brackets in the python script (see bugzilla issue).
> 
> Yep, I think I need a SME. Who would be good, do you think?

@admd recently helped me on a hub problem (with PR uyuni-project/uyuni#2191) so I guess he's a good candidate :smile_cat:  .

**Comment by admd** [comment]
@Loquacity 

Removing the angle brackets from all there examples here https://github.com/uyuni-project/uyuni-docs/blob/master/modules/large-deployments/pages/hub-auth.adoc should be enough

### Before
```
client = xmlrpclib.Server(<HUB_URL>, verbose=0)

hubKey = client.hub.login(<HUB_LOGIN>, <HUB_PASSWORD>)

```
### After
```
client = xmlrpclib.Server(HUB_URL, verbose=0)

hubKey = client.hub.login(HUB_LOGIN, HUB_PASSWORD)

```

**Comment by Loquacity** [comment]
> @Loquacity
> 
> Removing the angle brackets from all there examples here https://github.com/uyuni-project/uyuni-docs/blob/master/modules/large-deployments/pages/hub-auth.adoc should be enough
> 
> ### Before
> ```
> client = xmlrpclib.Server(<HUB_URL>, verbose=0)
> 
> hubKey = client.hub.login(<HUB_LOGIN>, <HUB_PASSWORD>)
> ```
> 
> ### After
> ```
> client = xmlrpclib.Server(HUB_URL, verbose=0)
> 
> hubKey = client.hub.login(HUB_LOGIN, HUB_PASSWORD)
> ```

we use the angle brackets to indicate replaceables, though. those things still need to be replaced, right?

**Comment by admd** [comment]
Ahan I see. Yeah, customer needs to replace the values for these variables. 

```
HUB_URL = "http://localhost:8888/hub/rpc/api"
HUB_LOGIN = "admin"
HUB_PASSWORD = "admin"
```
Maybe we should place brackets around values then?

**Comment by Loquacity** [comment]
> Ahan I see. Yeah, customer needs to replace the values for these variables.
> 
> ```
> HUB_URL = "http://localhost:8888/hub/rpc/api"
> HUB_LOGIN = "admin"
> HUB_PASSWORD = "admin"
> ```
> 
> Maybe we should place brackets around values then?

Any value that needs to be replaced should be denoted by angle brackets, that's how we do all code snippets/examples throughout the doc. To some extent, it *should* break the code if they get so far as to actually try and run it without replacing the value.

File: modules/large-deployments/pages/hub-auth.adoc [hunk 1]
```diff
@@ -50,23 +50,23 @@ In this case, the response will be a single value and not a map, in the same way
...
 
-hubKey = client.hub.login(<HUB_LOGIN>, <HUB_PASSWORD>)
+hubKey = client.hub.login(HUB_LOGIN, HUB_PASSWORD)
 
-#Get the server IDs
+# Get the server IDs
 serverIds = client.hub.listServerIds(hubKey)
 
-# authenticate those servers(each server with it's own credentials)
+# Authenticate those servers(each server with it's own credentials)
```
Comment (keichwa) [inline modules/large-deployments/pages/hub-auth.adoc:64]:
```suggestion
# Authenticate each server with its own credentials
```

File: modules/large-deployments/pages/hub-auth.adoc [hunk 2]
```diff
@@ -50,23 +50,23 @@ In this case, the response will be a single value and not a map, in the same way
...
 usernames = ["admin" for s in serverIds]
 passwords = ["admin" for s in serverIds]
 client.hub.attachToServers(hubKey, serverIds, usernames, passwords)
 
-# perform the needed operation
+# Perform the needed operation
```
Comment (keichwa) [inline modules/large-deployments/pages/hub-auth.adoc:69]:
```suggestion
# Perform the operation
```

**Comment by Bischoff** [comment]
It's much better so now.

Of course this code will still break, but at least the replaceables are conveniently isolated on top of script now and I think the reader can easily identify them as such.

Also, removing our internal test values was a good thing.

File: modules/large-deployments/pages/hub-auth.adoc [hunk 3]
```diff
@@ -50,23 +50,23 @@ In this case, the response will be a single value and not a map, in the same way
...
+hubKey = client.hub.login(HUB_LOGIN, HUB_PASSWORD)
 
-#Get the server IDs
+# Get the server IDs
 serverIds = client.hub.listServerIds(hubKey)
 
-# authenticate those servers(each server with it's own credentials)
+# Authenticate each server with its own credentials
 usernames = ["admin" for s in serverIds]
 passwords = ["admin" for s in serverIds]
```
Comment (Bischoff) [inline modules/large-deployments/pages/hub-auth.adoc:66]: Change also these two `"admin"` strings into variables defined above (again with `<REPLACEABLE>`s)

**Comment by keichwa** [comment]
On 2020-05-18 09:17, E Bischoff wrote:

> Of course this code will still break, but at least the replaceables
> are conveniently isolated on top of script now and I think the reader
> can easily identify them as such.

:+1: !

-- 
Karl Eichwalder                    SUSE Software Solutions Germany GmbH
                                    Maxfeldstraße 5
                                    90409 Nürnberg, Germany
Geschäftsführer: Felix Imendörffer (HRB 36809, AG Nürnberg)

**Comment by Bischoff** [comment]
Thanks for last changes @Loquacity ! :smiley:

**Pull request merged**